lib/kitchen/provisioner/ansible_push.rb

```diff
@@ -88,7 +89,11 @@ def conf
 
       def machine_name
         return @machine_name if defined? @machine_name
-        @machine_name = instance.name.gsub(/[<>]/, '').split("-").drop(1).join("-")
+        unless config[:use_instance_name]
+          @machine_name = instance.platform.name.gsub(/[<>]/, '')
```

Comment:
I think using platform name might be an issue. As it translates **-** to **_** for example. While probably its safer than using 
```ruby
instance.name.gsub(/[<>]/, '').split("-").drop(1).join("-")
```
But that would be breaking changes a version upgrade might be required.  Another aproach would be just to use the old function and in the next breaking change would include that your suggestion ?

Whats your thoughts ?